pkg/systemd/logs.js

```diff
@@ -414,48 +585,95 @@ $(function() {
         var tab = $('<ul class="nav nav-tabs nav-tabs-pf">');
 
         var d_btn = $('<button class="btn btn-danger problem-btn btn-delete pficon pficon-delete">');
-        var r_btn = $();
-        if (problem.IsReported) {
-            for (var pid = 0; pid < problem.Reports.length; pid++) {
-                if (problem.Reports[pid][0] === 'ABRT Server') {
-                    var url = problem.Reports[pid][1]['URL']['v']['v'];
-                    r_btn = $('<a class="problem-btn">')
-                            .attr('href', url)
-                            .attr("target", "_blank")
-                            .text(_("Reported"));
-                    break;
-                }
-            }
-        } else if (problem.CanBeReported) {
-            r_btn = $('<button class="btn btn-primary problem-btn">').text(_("Report"));
-
-            r_btn.click(function() {
-                tab.children(':last-child').replaceWith($('<div class="spinner problem-btn">'));
-                var proc = cockpit.spawn(['reporter-ureport', '-d', problem.ID], { superuser: 'true' });
-                proc.done(function() {
-                    window.location.reload();
-                });
-                proc.fail(function(ex) {
-                    var message;
-                    // 70 is 'This problem has already been reported'
-                    if (ex.exit_status === 70) {
-                        window.location.reload();
-                        return;
-                    } else if (ex.problem === 'access-denied') {
-                        message = _("Not authorized to upload-report");
-                    } else if (ex.problem === "not-found") {
-                        message = _("Reporter 'reporter-ureport' not found.");
-                    } else {
-                        message = _("Reporting was unsucessful. Try running `reporter-ureport -d " + problem.ID + "`");
-                    }
-                    $('<div class="alert alert-danger">')
-                            .append('<span class="pficon pficon-error-circle-o">')
-                            .text(message)
-                            .insertAfter(".breadcrumb");
-                    tab.children(':last-child').replaceWith($('<span>'));
+        var r_btn = $('<button class="btn btn-primary" disabled="">').text(_("Report"));
+        var r_btn_div = $('<div class="btn-group problem-btn"></div>');
+
+        var badge = $('<span class="badge" style"vertical-align:baseline;"></span>');
+
+        r_btn_div.append(r_btn);
+        r_btn_div.css('padding', '0');
+
+        badge.text(_('Reported'));
+
+        reportd.wait()
+                .done(function() {
+                    reportd.GetWorkflows(problem.path).done(function(args, options) {
+                        if (args.length > 1) {
+                            var menu = $('<ul class="dropdown-menu dropdown-menu-right" role="menu"></ul>');
```

Comment:
There is inconsistent usage of let/var and from what I see they are used interchangeably in this patch. Can you please make these a bit consistent?

Reply:
Can you provide specific instances? Such broad comments really don’t help me.

Reply:
The idea is to have logical scoping, so I tried to only use `var` when the variable was needed in a callback. It might have gotten out of hand due to several rewrites, however.

Reply:
So if you used them deliberately it's fine. I just wanted to make sure that it's not random, since I don't know your background with js :)

Reply:
In that case - my background is nearly nonexistent, but most of the things were done deliberately. Maybe sometimes uninformedly, but deliberately. Some things (like vertical whitespace, code layout that reeks of C-isms) I will not try to defend, because I don’t have the energy to.

</ramble>

Reply:
Hope these nit picks didn't discourage you. Please ignore, and please post some screenshots of the new feature + some tests to be able to do some more to the point review ;)

Thanks!